src/components/content/credential/Credential.tsx

```diff
@@ -126,7 +126,7 @@ function Credential(): JSX.Element {
 
     const updateCredential = (abstractCredentialInfo: AbstractCredentialInfo) => {
         const credentialVariables: CredentialVariables = abstractCredentialInfo;
-        const userName: string | null = localStorage.getItem(usernameKey);
+        const userName: string | null = sessionStorage.getItem(userRoleKey);
```

Comment:
here also we must get the username from token. Not the  role name. 
Role in session storage must be used only for setting the previous selected role. For nothing else.